### ISSUE TITLE
Temporary file upload fails when using "non string castable" properties in the component

### ISSUE THREAD

**Issue by gregoriohc** [livewire/livewire#1073]
### Description
New upcoming file upload feature fails if you are using public component with properties other than anything that could be converted to a string (an array or a model for example).
In the documentation, it says: "Properties can ONLY be either a JavaScript-friendly data types (string, int, array, boolean), OR an eloquent model (or collection of models)."
So, maybe it could be necessary to check if the subject is a string in the "canUnserialize" method of the "TemporarilyUploadedFile" class? Another option it's to check it in the "hydratePropertyFromWithFileUploads" method of the "WithFileUploads" trait.

### Error
```php
ErrorException: strncmp() expects parameter 1 to be string, array given
```

### Expected behaviour
Upload temporary file and make it available in the component property

### Steps to reproduce
Create a component with the new "WithFileUploads" trait, a file property, and some other properties that can't be casted to a string:
```php
class ComponentWithFileUploads extends Component
{
    public $photo;
    public $nonStringProperty = ['foo', 'bar'];

    public function render()
    {
        return '<div><input type="file" wire:model="photo" /></div>';
    }
}
```

**Context**
- Livewire version: dev-master
- Laravel version: 7.14.1
- Browser: Chrome

**Comment by gregoriohc** [comment]
This is related with #958

**Comment by gregoriohc** [comment]
I saw that this has been fixed in the 1.2 release, so I'm closing this. Thanks!
